File: app/src/main/java/de/dizcza/fundu_moto_joystick/fragment/SonarView.java
package de.dizcza.fundu_moto_joystick.fragment;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Path;
import android.util.AttributeSet;
import android.view.View;

public class SonarView extends View {
    private static final int POINT_RADIUS = 10;
    private final Paint mPaint = new Paint();
    private final Path mPath = new Path();

    public SonarView(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
        mPaint.setStyle(Paint.Style.FILL);
        mPaint.setColor(Color.BLACK);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        canvas.drawPath(mPath, mPaint);
    }

    public void drawCircle(float x, float y) {
        mPath.addCircle(x, y, POINT_RADIUS, Path.Direction.CW);
        invalidate();
    }

    public void clear() {
        mPath.reset();
        invalidate();
    }
}

File: app/src/main/java/de/dizcza/fundu_moto_joystick/fragment/SonarView.kt
package de.dizcza.fundu_moto_joystick.fragment

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.Path
import android.util.AttributeSet
import android.view.View

class SonarView(context: Context?, attributeSet: AttributeSet?) : View(context, attributeSet) {
    private val mPaint = Paint()
    private val mPath = Path()

    init {
        mPaint.style = Paint.Style.FILL
        mPaint.color = Color.BLACK
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        canvas.drawPath(mPath, mPaint)
    }

    fun drawCircle(x: Float, y: Float) {
        mPath.addCircle(x, y, POINT_RADIUS.toFloat(), Path.Direction.CW)
        invalidate()
    }

    fun drawRect(x: Float, y: Float) {
        mPath.addRect(x - RECT_SIZE, y - RECT_SIZE, x + RECT_SIZE, y + RECT_SIZE, Path.Direction.CW)
        invalidate()
    }

    fun clear() {
        mPath.reset()
        invalidate()
    }

    companion object {
        private const val POINT_RADIUS = 10
        private const val RECT_SIZE = 20
    }
}
